### PR TITLE
Long-read SV support with Sniffles 2 (#263)

### PR DESCRIPTION
**Pull request by holtgrewe**

### CODE REVIEW

**Comment by holtgrewe** [comment]
@eudesbarbosa @Nicolai-vKuegelgen this one is ready for review

File: snappy_pipeline/workflows/ngs_mapping/Snakefile [hunk 1]
```diff
@@ -169,7 +169,7 @@ rule ngs_mapping_minimap2_run:
     output:
         **wf.get_output_files("minimap2", "run"),
     threads: wf.get_resource("minimap2", "run", "threads")
-    resources:
```
Comment (eudesbarbosa) [inline snappy_pipeline/workflows/ngs_mapping/Snakefile:172]: This looks off. Should it be something like below...
```
    resources:
        time=wf.get_resource("minimap2", "run", "time"),
        memory=wf.get_resource("minimap2", "run", "memory"),
        partition=wf.get_resource("minimap2", "run", "partition"),
    params:
        args=wf.substep_dispatch("minimap2", "get_args", "run"),

```

File: snappy_pipeline/workflows/ngs_mapping/__init__.py [hunk 2]
```diff
@@ -663,10 +663,17 @@ def get_resource_usage(self, action):
         mem_gb = int(3.5 * self.config["minialign"]["mapping_threads"])
         return ResourceUsage(
             threads=self.config["minialign"]["mapping_threads"],
-            time="4-00:00:00",  # 4 days
+            time="0-12:00:00",  # 2 days
```
Comment (eudesbarbosa) [inline snappy_pipeline/workflows/ngs_mapping/__init__.py:666]: Cosmetic: comment is wrong.

File: snappy_pipeline/workflows/wgs_sv_calling/__init__.py [hunk 3]
```diff
@@ -92,7 +92,7 @@
 
 - ``"dna_long"`` (PacBio)
     - ``"pb_honey_spots"``
-    - ``"sniffles"``
+    - ``"sniffles"``, ``"sniffles2"``
```
Comment (eudesbarbosa) [inline snappy_pipeline/workflows/wgs_sv_calling/__init__.py:95]: Maybe new line to keep it consistent with the rest?

File: snappy_pipeline/workflows/wgs_sv_calling/__init__.py [hunk 4]
```diff
@@ -169,6 +170,9 @@
       num_threads: 16
     sniffles:
       num_threads: 16
+    sniffles2:
+      num_threads: 16
+      tandem_repeats: /fast/groups/cubi/work/projects/biotools/sniffles2/trf/GRCh37/human_hs37d5.trf.bed
```
Comment (eudesbarbosa) [inline snappy_pipeline/workflows/wgs_sv_calling/__init__.py:175]: Probably tag it with 'REQUIRED' so it goes to local config?

File: snappy_wrappers/wrappers/minimap2/wrapper.py [hunk 5]
```diff
@@ -8,7 +8,12 @@
 
 this_file = __file__
 
-# TODO: write out separate read groups
+seq_platform = snakemake.params.args["extra_infos"]["seqPlatform"]
+library_kit = snakemake.params.args["extra_infos"]["libraryKit"]
+
+import pdb
+
+pdb.set_trace()
```
Comment (eudesbarbosa) [inline snappy_wrappers/wrappers/minimap2/wrapper.py:16]: Construction leftover?

**Review by eudesbarbosa — CHANGES_REQUESTED** [review]
Looks good. I marked a few things for clarification or cosmetic reasons.

**Comment by holtgrewe** [comment]
Should be fixed now.

**Review by eudesbarbosa — APPROVED** [review]
LGTM

**Pull request merged**